### ISSUE TITLE
Unable to open Oni2 on mac os 10.15.6

### ISSUE THREAD

**Issue by mohitt** [onivim/oni2#2225]
I tried opening Oni2 on mac using finder. I can see the icon in the app switcher but no window. In `Activity Monitor` it shows up as not responding.

**Comment by mohitt** [comment]
I have brought the early access lic as well.

**Comment by bryphe** [comment]
Hi @mohitt ,

Sorry about the issue launching, thanks for reporting (and for supporting the project!)

A few questions:
- What sort of hardware are you using? Is it a Macbook, Mac Mini?
- Would it be possible to grab the log files? You can run this command from `Terminal`: ` /Applications/Onivim2.app/Contents/MacOS/Oni2 -f --debug --log-file=onivim.log` which will give some output. Might give us some clues about where it's hanging.
- Are you using multiple monitors? Or are you using the 'display scaling' setting? It might be another manifestation of: https://github.com/onivim/oni2/issues/1185

**Comment by bryphe** [comment]
Thanks for the details, @mohitt ! Much appreciated

Indeed, from the log:
```
[DEBUG]    +0ms Revery.Core.Window : Starting window creation...
[INFO]     +0ms Revery.Core.Window : Creating window Oni2 width: 800 height: 600
[INFO]    +19ms Revery.Core.Window : Window created successfully.
[DEBUG]    +0ms Revery.Core.Window : - Id: 1
[DEBUG]    +0ms Revery.Core.Window : - PixelFormat: SDL_PIXELFORMAT_ARGB8888
[DEBUG]    +0ms Revery.Core.Window : Setting window context
[ERROR] +361.0s Revery.SDL2 : SDL_GL_Setup failed: Failed creating OpenGL pixel format
[DEBUG]    +0ms Revery.Core.Window : GL setup. Checking GL version...
[DEBUG]    +0ms Revery.Core.Window : Checking GL vendor...
[DEBUG]    +0ms Revery.Core.Window : Checking GL shading language version...
[INFO]     +0ms Revery.Core.Window : OpenGL hardware info:
[INFO]     +0ms Revery.Core.Window :   renderer: (null)
[INFO]     +0ms Revery.Core.Window :   version: (null)
[INFO]     +0ms Revery.Core.Window :   vendor: (null)
[INFO]     +0ms Revery.Core.Window :   shadingLanguageVersion: (null)
```
Looks related to #1185 

I've been trying to track that one down for a while!

Would it be possible to see the system info - `Overview` and `Displays`?

![image](https://user-images.githubusercontent.com/13532591/89452937-50501f00-d713-11ea-9a46-301e878db349.png)

And the 'Overviews' section (serial number not needed, mainly the 'Graphics' is what I'm looking for)
![image](https://user-images.githubusercontent.com/13532591/89452964-5ba34a80-d713-11ea-926f-cf46fe8cbc38.png)

As well as the 'Displays' - showing the monitors along with the graphics card being used:
![image](https://user-images.githubusercontent.com/13532591/89453012-707fde00-d713-11ea-8ed9-8d153de317e5.png)

Thanks again for the help and details-

**Comment by mohitt** [comment]
This laptop has dual gpu. and 4 display connected using DisplayLink usb docking station. 

![Screen Shot 2020-08-05 at 1 54 43 PM](https://user-images.githubusercontent.com/919686/89463229-7a114200-d723-11ea-87f8-1d9c8a534305.png)


![Screen Shot 2020-08-05 at 1 53 45 PM](https://user-images.githubusercontent.com/919686/89463247-809fb980-d723-11ea-88c7-472aafe6bd96.png)

**Comment by bryphe** [comment]
Hi @mohitt - thanks a lot for the details! Very helpful.

I'm investigating the issue (and working on borrowing / procuring a Macbook with a dual-GPU set-up to test on). I suspect we'll need the following:

- Set the `NSSupportsAutomaticGraphicsSwitching` in the `plist` file - this is related to #2090
- Add logging to Revery around the available displays / adapter settings
- Some flags here might need to be changed: https://github.com/revery-ui/revery/blob/1b06ef179fa5fa0ea3d405f819d4f67cc5c4410b/packages/reason-sdl2/src/sdl2_wrapper.cpp#L1362 (affects the choice of renderer / display)

Hopefully should have a device mid next week to reproduce and fix this and #1185 🤞

**Comment by taylorthurlow** [comment]
I can't test this right now because the machine is actually being repaired by Apple, but I've used semi-recent (last month or so?) builds of oni2 on my Macbook Pro 16", on 10.15.5. I also run a docking station with 3 monitors and a closed laptop lid - I have no issues opening oni2 on that machine. I can double check when I have it on hand again, but I just wanted to chime in as I have not experienced this issue.

The 16" and 15" are obviously different machines, though.

**Comment by bryphe** [comment]
Along with #1185, this should be fixed now with #2278 and #2283 (which picked up https://github.com/revery-ui/revery/pull/984)

Please let me know if you're still seeing this, though-
